torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -0,0 +1,215 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/freeze_module.h>
+
+#include <torch/csrc/jit/graph_executor_impl.h>
+#include <torch/csrc/jit/passes/inliner.h>
+
+#include <stack>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+class AttributePropagator {
```

Comment:
Nit: the class name was a bit confusing to me: the file is about freezing, but the name is about some propagation.

Reply:
I see. Freezing allows folding attribute :). I guess... I keep the name.